apps/site/lib/site_web/controllers/customer_support_controller.ex

```diff
@@ -41,6 +79,22 @@ defmodule SiteWeb.CustomerSupportController do
     end
   end
 
+  @spec render_expandable_blocks(Plug.Conn.t(), list) :: [Phoenix.HTML.safe()]
+  def render_expandable_blocks(assigns, content_blocks \\ @content_blocks) do
+    content_blocks
+    |> Enum.map(fn block ->
+      SiteWeb.PartialView.render(
+        "_expandable_block.html",
+        Map.merge(assigns, %{
+          header: block.header,
+          id: block.id,
+          initially_expanded: block.initially_expanded,
+          view_template: "_#{block.id}.html"
```

Comment:
This is so cool!

One thing I wonder is what would happen if someone passed in a block.id where the corresponding view_template doesn't exist. Would that fail gracefully? 🤔

Reply:
Good point! I made some changes so at least the page can still be viewed but the blocks that don't have a template are just not rendered.